### DISCUSSION TITLE
load img into waveform

### DISCUSSION THREAD

**Discussion by Elgoritm** [katspaugh/wavesurfer.js#2930] [Q&A]
Hey there!

I've been trying to load an image into the waveform, so far I only got to a black background. I've been trying to do the following: 

```
const img = new Image();
img.onload = function() {    
    ctx.drawImage(img, 0, 0);                
};                            
img.src = "/img/12.jpg"; 
```

Any help would be appreciated:)

**Comment by katspaugh** [comment]
From where are you doing it? A custom `renderFunction` param?

**Reply by Elgoritm** [reply]
I first tried doing a similar thing to the gradient example. I think this also got me to a black background.. 

```
const ctx = document.createElement('canvas').getContext('2d')
const gradient = ctx.createLinearGradient(0, 0, 0, 150)
gradient.addColorStop(0, 'rgb(200, 0, 200)')
gradient.addColorStop(0.7, 'rgb(100, 0, 100)')
gradient.addColorStop(1, 'rgb(0, 0, 0)')
```

like this: 
```
const img = new Image();                
img.src = "/img/12.jpg";  

const context = document.createElement('canvas').getContext('2d')
const gradient = context.drawImage(img, 0, 0)
```

ws.create etc

**Reply by katspaugh** [reply]
This doesn’t answer my question. You’re creating your own canvas. I don’t see how this is related to wavesurfer at all.

The gradient example shows how to set a gradient as the waveColor. Drawing an image file in the same way won’t work.

**Reply by Elgoritm** [reply]
I'm sorry. I think my understanding is very limited, including not understanding your question.. I thought it might be possible to insert an image instead of a gradient background. Am I correct that it is possible, but not in the same way as the gradient example?

If it is possible, how should I approach it?

**Reply by katspaugh** [reply]
Actually it is possible to set a CanvasPattern as waveColor.
You can create one from an image, see here: https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/createPattern

**Reply by katspaugh** [reply]
Here's an example:

```js
import WaveSurfer from 'https://unpkg.com/wavesurfer.js@beta'

// Create a canvas pattern
const image = new Image()
image.src = '/logo-small.png'
const ctx = document.createElement('canvas').getContext('2d')
const pattern = ctx.createPattern(image, 'repeat')

// Set the pattern as waveColor
WaveSurfer.create({
  container: document.body,
  waveColor: pattern,
  progressColor: 'rgba(0, 0, 100, 0.5)',
  url: '/examples/audio/audio.wav',
})
```

You can paste it on https://wavesurfer-js.org/examples

**Reply by Elgoritm** [reply]
This works like a charm in the example. Thank you for your help, I really appreciate it!

Somehow if I implement, I still get a black background. I saved the logo-small.png and am loading it into my project. 

Can it have anything to do with working with vite?

**Reply by katspaugh** [reply]
It might have to do with the server. Try first just rendering the image on its own as an HTML tag. CORS should be enabled for it and it should be on the same domain.

**Reply by Elgoritm** [reply]
loading it in the html does work like expected...
